edison-c.js

```diff
@@ -110,8 +132,14 @@ function initTasks(gulp, options) {
 
   gulp.task('run-internal', false, function (cb) {
     var param = options.appParams || '';
-    all.sshExecCmd('sudo chmod +x ' + targetFolder + '/' + startFile + ' ; sudo '
-      + targetFolder + '/' + startFile + ' ' + param, { verbose: true, sshPrintCommands: true }, cb);
+
+    var dict = createDictionary(config.iot_device_connection_string, ';');
+    if (dict['x509'] && dict['x509'] === 'true') {
+      param += ' "x509=true"';
```

Comment:
>param += ' "x509=true"'; [](start = 6, length = 24)

Device connection string is being passed as a parameter already, correct? Is this to avoid parsing it in the C client code?